src/ansi-c/expr2c.cpp

```diff
@@ -3425,6 +3425,31 @@ std::string expr2ct::convert_sizeof(
   return dest;
 }
 
+/// Convert a let expression from an irep to a string, taking into account
+/// precedence.
+/// \param src: The let expression to be converted.
+/// \param precedence: The precedence to be obeyed during conversion.
+/// \return: A string representation of the expression.
+std::string expr2ct::convert_let(const exprt &src, unsigned &precedence)
+{
+  const auto &let_expr = to_let_expr(src);
+  std::string dest = "let ";
+  dest += convert_with_precedence(let_expr.symbol(), precedence);
+  dest += " = ";
+  dest += convert_with_precedence(let_expr.value(), precedence);
+  dest += " in (";
+  dest += convert_with_precedence(let_expr.where(), precedence);
+  dest += ")";
+  return dest;
+}
```

Comment:
As already pointed out by Michael, I don't think that the pretty printer for LET expressions belongs into the C front-end. Note that format_expr can already pretty print LET expressions.